kitsune/wiki/models.py

```diff
@@ -209,15 +210,30 @@ def _ensure_inherited_attr(self, attr):
                 self.translations.all().update(**{attr: getattr(self, attr)})
 
     def _clean_category(self):
-        """Make sure a doc's category is the same as its parent's."""
+        """Make sure a doc's category is valid."""
         if (not self.parent and
                 self.category not in (id for id, name in CATEGORIES)):
             # All we really need to do here is make sure category != '' (which
             # is what it is when it's missing from the DocumentForm). The extra
             # validation is just a nicety.
             raise ValidationError(_('Please choose a category.'))
+
         self._ensure_inherited_attr('category')
 
+    def _clean_template_status(self):
+        if (self.category == TEMPLATES_CATEGORY and
+                not self.title.startswith(TEMPLATE_TITLE_PREFIX)):
+            # L10n: Note, do not translate "Template:" in this string.
+            raise ValidationError(_(u'Documents in the Template category must have titles that '
+                                    u'start with "Template:". (Current title is "{}")'
+                                    .format(self.title)))
+
+        if self.title.startswith(TEMPLATE_TITLE_PREFIX) and self.category != TEMPLATES_CATEGORY:
+            # L10n: Note, do not translate "Template:" in this string.
+            raise ValidationError(_(u'Documents with titles that start with "Template:" must be '
```

Comment:
A big +1 to this and all the other places we should use the constant instead of a string.